### PR TITLE
Don't rebuild web assets if nothing has changed.

### PR DESCRIPTION
**Pull request by nullstalgia**
Untested in Windows.

Also, what was the function of the `Import("env")` ?

Couldn't find anything like it, and it seems fine without it :o

### CODE REVIEW

**Comment by sidoh** [comment]
Great idea.  Have wanted this myself.

`Import("env")` is something from SCons.  It allows you to interact with the [platformio build](https://docs.platformio.org/en/latest/projectconf/advanced_scripting.html#construction-environments) programmatically.  Don't think it's required here because the interface between the build and the script are purely args/stdout.

File: .build_web.py [hunk 1]
```diff
@@ -31,6 +34,13 @@ def build_web():
                 os.mkdir("../dist")
 
             copyfile("build/web_assets.h", "../dist/web_assets.h")
+
+            generated_timestamp = "//-"+str(int(time()))+"-"
+            with open("../dist/web_assets.h", 'r+') as f:
+                content = f.read()
+                f.seek(0, 0)
+                f.write(generated_timestamp + '\n' + content)
+
```
Comment (sidoh) [inline .build_web.py]: Could we not just compare `st_mtime` of the `web_assets` file from `os.stat` instead of dumping/parsing it?

Comment (nullstalgia) [reply .build_web.py]: I think I had a reason for this, but I can't remember to be perfectly honest. 3am code isn't the most logical at times. :sweat_smile: 

I'll swap it to just the file's `st_mtime` unless either of us can find a reason to keep it in.

Besides, the script should be cleaner without it.

Comment (sidoh) [reply .build_web.py]: Awesome, thanks :)

**Comment by sidoh** [comment]
I'm gonna test this out on Windows before merging -- should have time tomorrow or over the weekend.

**Comment by sidoh** [comment]
Looks like it works in Windows!

Had to complete a bit of a side-quest to get Windows build to work: PlatformIO was installing both a patched version of the Time dependency and the published one.  The published one has a file called `time.h` which conflicts with an SDK dependency and wreaks havoc.  Deleting `.pio/libdeps/Time_ID44` does the trick.

Looks like there are pretty good native methods for doing time stuff these days, so the right move is probably to switch off of this library.

**Comment by sidoh** [comment]
Thanks, @nullstalgia!

**Pull request merged**